### PR TITLE
ci: Export GitHub Actions Workflows traces to Honeycomb.io

### PR DESCRIPTION
**Pull request by gerhard**
The goal is for us to better understand where our workflows spend time, including:
- How long are jobs waiting for available runners?
- How long do steps before & after Dagger take?
- How do we check that Dagger Cloud reporting is accurate?

The last one is especially important to me since it's directly related to **always run 2 of everything**, where Dagger Cloud is 1/2, and this would be 2/2 (a check for the check).

Without this addition, it's very difficult to know where our workflows as a whole spend most time, and therefore where we should focus our next optimisations.

---
Having wired this up in my fork - a new workflow needs to run on the default repo branch before it gets recognised - this is what the end-result looks like:

image

---
This is the view that I really wanted us to have:
image

---
I wanted this view for so long, and now finally have it! Note, this is using the free runners. Once we merge it, we will see how our `dagger-runners` compare:
image

---
There are bunch more things that will stem out of this, but for now this gives us everything that we need to better understand the health of our GitHub Actions workflows.

Follow-ups:
- [ ] Create a read-only account & share with the team
- [ ] Build & share some queries that track the total workflow execution over time
- [ ] When HTTPS support is added, we can switch to Tempo on Grafana Cloud https://github.com/inception-health/otel-export-trace-action/issues/194 (this PR introduces a new tool to our infra landscape)

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -152,12 +152,12 @@ jobs:
       - name: "Setup Rust"
         uses: dtolnay/rust-toolchain@stable
 
-      - name: "ALWAYS print engine logs - especially useful on failure"
-        if: always()
+      - name: "Print Engine logs on failure"
```
Comment (jedevc) [inline .github/workflows/publish.yml:155]: IMO, we should keep these to always if we can.

I've found it really useful to go back through old successful runs and compare logs to new failing runs.

**Review by jedevc — APPROVED** [review]
:tada: this looks awesome! Just one small comment!

File: .github/workflows/publish.yml (hunk 1 above)
Comment (gerhard) [reply .github/workflows/publish.yml:155]: OK, will change this before merging.

**Comment by gerhard** [comment]
Thank you for the review & 👍

Will change back to `always()` before merging.

**Pull request merged**